cortix/src/network.py

```diff
@@ -225,60 +227,38 @@ def run(self):
             if self.rank != 0:
                 mod = self.modules[self.rank-1]
                 self.log.info('Launching Module {}'.format(mod))
-                mod.run()
+                mod.run_and_save()
 
             # Sync here at the end
             self.comm.Barrier()
 
-            if len(self.modules) > self.max_n_modules_for_data_copy_on_root:
```

Comment:
Could you leave the in-memory copy for now as an alternative and also so the examples and notebook are not broken? I don't have time to fix them now and it does not hurt to have both options: in-core and out-of-core saves.
If you run the examples and the notebook, they should all pass as they are at the moment after you added the out-of-core save. 
Thanks.